### PR TITLE
Implement DeployToUpgrade.

### PR DESCRIPTION
**Pull request by phrohdoh**
#7653 v2

### CODE REVIEW

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+		public void ResolveOrder(Actor self, Order order)
+		{
+			if (order.OrderString != "DeployToUpgrade")
+				return;
+
+			if (info.AllowedTerrainTypes.Length > 0)
+			{
+				var world = self.World;
+				var map = world.Map;
+				var pos = map.CellContaining(self.CenterPosition);
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: Use `self.Location` instead.

File: OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+				var tileSet = world.TileSet;
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Upgrades/DeployToUpgrade.cs]: These variables are only used once or twice, so it doesn't seem worth having them all.

**Comment by Phrohdoh** [comment]
Updated.

**Comment by penev92** [comment]
I want to note while this is still in my head:
If you have an actor with a set of valid terrain types, you will get the "deploy" cursor even when standing on invalid terrain. Thumpers are the first test case that comes to mind.

**Comment by penev92** [comment]
I'm not going to argue over this. That's why I pushed a quick fix to [this branch](https://github.com/penev92/OpenRA/tree/test_7865_deployToUpgrade). It also contains a small testcase. The upgrade itself doesn't work, but that is because I picked one at random. The cursor is important here.
I suggest you take the changes you need (minus the test case).

**Comment by Phrohdoh** [comment]
Done.

**Comment by penev92** [comment]
Looks good to me now :+1:

**Comment by pchote** [comment]
Code looks good, and I don't see any obvious regressions ingame :+1:

**Comment by pchote** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-%28bleed%29/_compare/109037fdd60c71973723948fb602eed6de4e9512%5E...109037fdd60c71973723948fb602eed6de4e9512

**Pull request merged**